Build fail on 32bit platforms like Rasp3b

It seems size_t is the same type as uint32_t on RPi, and causing build fails. 

```
treelite/include/treelite/common.h:372:15: error: redefinition of ‘T treelite::common::TextToNumber(const string&) [with T = unsigned int; std::__cxx11::string = std::__cxx11::basic_string<char>]’
 inline size_t TextToNumber(const std::string& str) {
               ^~~~~~~~~~~~
/home/pi/workplace/neo-ai/neo-ai-dlr/3rdparty/treelite/include/treelite/common.h:357:17: note: ‘T treelite::common::TextToNumber(const string&) [with T = unsigned int; std::__cxx11::string = std::__cxx11::basic_string<char>]’ previously declared here
 inline uint32_t TextToNumber(const std::string& str) {
                 ^~~~~~~~~~~~

```

--- a/include/treelite/common.h
+++ b/include/treelite/common.h
@@ -288,9 +288,9 @@ inline T TextToNumber(const std::string& str) {
                 || std::is_same<T, int>::value
                 || std::is_same<T, int8_t>::value
                 || std::is_same<T, uint32_t>::value
-                || std::is_same<T, size_t>::value,
+                || std::is_same<T, uint64_t>::value,
                 "unsupported data type for TextToNumber; use float, double, "
-                "int, int8_t, uint32_t, or size_t.");
+                "int, int8_t, uint32_t, or uint64_t.");
 }
 
 template <>
@@ -369,18 +369,18 @@ inline uint32_t TextToNumber(const std::string& str) {
 }
 
 template <>
-inline size_t TextToNumber(const std::string& str) {
+inline uint64_t TextToNumber(const std::string& str) {
   errno = 0;
   char *endptr;
   auto val = std::strtoull(str.c_str(), &endptr, 10);
-  if (errno == ERANGE || val > std::numeric_limits<size_t>::max()) {
+  if (errno == ERANGE || val > std::numeric_limits<uint64_t>::max()) {
     LOG(FATAL) << "Range error while converting string to size_t";
   } else if (errno != 0) {
     LOG(FATAL) << "Unknown error";
   } else if (*endptr != '\0') {
     LOG(FATAL) << "String does not represent a valid integer";
   }
-  return static_cast<size_t>(val);
+  return static_cast<uint64_t>(val);
 }
 
 /*!

--- a/src/frontend/lightgbm.cc
+++ b/src/frontend/lightgbm.cc
@@ -49,7 +49,7 @@ struct LGBTree {
   int num_cat;  // number of categorical splits
   std::vector<double> leaf_value;
   std::vector<int8_t> decision_type;
-  std::vector<size_t> cat_boundaries;
+  std::vector<uint64_t> cat_boundaries;
   std::vector<uint32_t> cat_threshold;
   std::vector<int> split_feature;
   std::vector<double> threshold;
@@ -212,7 +212,7 @@ inline treelite::Model ParseStream(dmlc::Stream* fi) {
       CHECK(it != dict.end() && !it->second.empty())
         << "Ill-formed LightGBM model file: need cat_boundaries";
       tree.cat_boundaries
-        = treelite::common::TextToArray<size_t>(it->second, tree.num_cat + 1);
+        = treelite::common::TextToArray<uint64_t>(it->second, tree.num_cat + 1);
       it = dict.find("cat_threshold");
       CHECK(it != dict.end() && !it->second.empty())
         << "Ill-formed LightGBM model file: need cat_threshold";

